### PR TITLE
W-10882348: (fix) dash as scalar is rendered with double quote now

### PR DESCRIPTION
**Pull request by tomsfernandez**

### CODE REVIEW

File: shared/src/test/scala/org/mulesoft/yaml/DocRenderTest.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.mulesoft.yaml
+
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import org.yaml.model.YDocument
+import org.yaml.render.YamlRender
+
+class DocRenderTest extends AnyFunSuite with Matchers {
```
Comment (pope1838) [inline shared/src/test/scala/org/mulesoft/yaml/DocRenderTest.scala:8]: YamlDocBuilderTest?

File: shared/src/main/scala/org/yaml/render/ScalarRender.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ object ScalarRender {
     if (l == 0)
       return if (mustBeString || !mark.plain) DoubleQuoteMark else NoMark
 
-    if (text.head == ' ' || text.endsWith("\n\n")) return DoubleQuoteMark
+    if (text.head == ' ' || text.endsWith("\n\n") || text == "-") return DoubleQuoteMark
```
Comment (pope1838) [inline shared/src/main/scala/org/yaml/render/ScalarRender.scala:57]: Please, extract to method like "shouldQuoteText" or "requireDobleQuote"

Comment (tomsfernandez) [reply shared/src/main/scala/org/yaml/render/ScalarRender.scala:57]: there are many things that require double quote in the code. Adding it just for 3 cases might be confusing for the reader

File: shared/src/test/scala/org/mulesoft/yaml/DocRenderTest.scala (hunk 1 above)
Comment (tomsfernandez) [reply shared/src/test/scala/org/mulesoft/yaml/DocRenderTest.scala:8]: I thought of that but that suite tests the DocBuilder specifically. I thought it wouldn't be 100% correct to put it there

**Review by looseale — APPROVED** [review]

**Pull request merged**